app/src/main/java/org/p2p/wallet/jupiter/statemanager/rate/SwapRateTickerManager.kt

```diff
@@ -19,27 +19,17 @@ import org.p2p.wallet.jupiter.repository.tokens.JupiterSwapTokensRepository
 import org.p2p.wallet.jupiter.statemanager.SwapCoroutineScope
 import org.p2p.wallet.jupiter.statemanager.SwapState
 import org.p2p.wallet.user.repository.UserLocalRepository
+import org.p2p.wallet.utils.divideSafe
 
 private const val TAG = "SwapRateTickerManager"
 
-class SwapRateTickerManager private constructor(
+class SwapRateTickerManager(
     swapScope: SwapCoroutineScope,
     private val userLocalRepository: UserLocalRepository,
     private val swapTokensRepository: JupiterSwapTokensRepository,
-    private val initDispatcher: CoroutineDispatcher
+    private val initDispatcher: CoroutineDispatcher = Executors.newSingleThreadExecutor().asCoroutineDispatcher()
```

Comment:
this dispatcher should be publicly settable, as unit tests should have the ability to override its behavior